File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int count=0;
    private int result=0;
    public void inorderTraversal(TreeNode root, int k){
        if(root==null){
            return;
        }
        inorderTraversal(root.left, k);
        count++;
        if(count==k){
            result=root.val;
            return;
        }
        inorderTraversal(root.right, k);
    }
    public int kthSmallest(TreeNode root, int k) {
        inorderTraversal(root, k);
        return result;
    }
}

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int count=0;
    int result=0;
    void inorderTraversal(TreeNode* root, int k){
        if(root==NULL){
            return;
        }
        inorderTraversal(root->left, k);
        count++;
        if(count==k){
            result=root->val;
            return;
        }
        inorderTraversal(root->right, k);
    }
    int kthSmallest(TreeNode* root, int k) {
        inorderTraversal(root, k);
        return result;
    }
};